src/SprykerEco/Zed/Payone/Business/Payment/MethodMapper/Klarna.php

```diff
@@ -0,0 +1,268 @@
+     */
+    protected function mapPaymentToAbstractAuthorization(
+        SpyPaymentPayone $paymentEntity,
+        AbstractAuthorizationContainer $authorizationContainer
+    ): ContainerInterface {
+        $authorizationContainer->setAid($this->getStandardParameter()->getAid());
+
+        $paymentDetailEntity = $paymentEntity->getSpyPaymentPayoneDetail();
+
+        $authorizationContainer->setAmount($paymentDetailEntity->getAmount());
+        $authorizationContainer->setCurrency($this->getStandardParameter()->getCurrency());
+        $authorizationContainer->setClearingType(PayoneApiConstants::CLEARING_TYPE_FINANCING);
+        $authorizationContainer->setFinancingType($paymentDetailEntity->getPayMethod());
+
+        $authorizationContainer->setReference($paymentEntity->getReference());
+
+        $paydataContainer = new PaydataContainer();
+        $paydataContainer->setAuthorizationToken($paymentDetailEntity->getTokenList());
+        $authorizationContainer->setPaydata($paydataContainer);
+
+        $personalContainer = $this->buildPersonalContainer($paymentEntity);
+        $authorizationContainer->setPersonalData($personalContainer);
+        $orderReference = $paymentEntity->getSpySalesOrder()->getOrderReference();
+        $authorizationContainer->setRedirect($this->createRedirectContainer($orderReference));
+
+        $shippingAddressEntity = $paymentEntity->getSpySalesOrder()->getShippingAddress();
+
+        if ($shippingAddressEntity) {
+            $shippingContainer = new ShippingContainer();
+            $this->mapShippingAddressToShippingContainer($shippingContainer, $shippingAddressEntity);
+        }
+
+        return $authorizationContainer;
+    }
+
+    /**
+     * @param \Orm\Zed\Payone\Persistence\SpyPaymentPayone $paymentEntity
+     *
+     * @return \SprykerEco\Zed\Payone\Business\Api\Request\Container\Authorization\PersonalContainer
+     */
+    protected function buildPersonalContainer(SpyPaymentPayone $paymentEntity): ContainerInterface
+    {
+        $personalContainer = new PersonalContainer();
+
+        $this->mapBillingAddressToPersonalContainer($personalContainer, $paymentEntity);
+
+        $currentRequest = $this->requestStack->getCurrentRequest();
+        $personalContainer->setIp($currentRequest->getClientIp());
+
+        return $personalContainer;
+    }
+
+    /**
+     * @param \SprykerEco\Zed\Payone\Business\Api\Request\Container\Authorization\PersonalContainer $personalContainer
+     * @param \Orm\Zed\Payone\Persistence\SpyPaymentPayone $paymentEntity
+     *
+     * @return void
+     */
+    protected function mapBillingAddressToPersonalContainer(
+        PersonalContainer $personalContainer,
+        SpyPaymentPayone $paymentEntity
+    ): void {
+        $orderEntity = $paymentEntity->getSpySalesOrder();
+        $billingAddressEntity = $orderEntity->getBillingAddress();
+        $personalContainer->setCountry($billingAddressEntity->getCountry()->getIso2Code());
```

Comment:
* What is the reason to redefine this method?

Reply:
This method doesn't set company. If company is set Klarna responses with error

Reply:
* Maybe better to do like that `$personalContainer = parent::mapBillingAddressToPersonalContainer(); $personalContainer->setCompany(null);`.

Reply:
++